### ISSUE TITLE
Won't build out-of-the-box; missing websocket-sharp

### ISSUE THREAD

**Issue by DanTup** [jammerware/margiebot#1]
I cloned the repo and tried to build in VS2015; but it failed:

[code]
1>C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets(1819,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "websocket-sharp". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
1>M:\Coding\TestApps\margiebot\MargieBot\Bot.cs(14,7,14,21): error CS0246: The type or namespace name 'WebSocketSharp' could not be found (are you missing a using directive or an assembly reference?)
1>M:\Coding\TestApps\margiebot\MargieBot\Bot.cs(37,17,37,26): error CS0246: The type or namespace name 'WebSocket' could not be found (are you missing a using directive or an assembly reference?)
[/code]

The reference to "websocket-sharp" in the `MargieBot` project doesn't seem to have a path, nor does it  appear to be a NuGet package.

**Comment by jammerware** [comment]
Hey, Danny,

Sorry you've having trouble. I did a bad job of making sure references made it into the repo for people cloning it. I just checked in a change that may fix it for you, so get latest and give it another try. If you still don't have any luck, try manually adding the reference - the DLL is in the Libs folder of the MargieBot.UI project.

Sorry for the trouble, I'm a little new at this. Let me know if you have any further questions.

Ben

**Comment by DanTup** [comment]
:+1:

I actually decided to have a go at [writing my own library](https://github.com/DanTup/SimpleSlackBot). My use case is pretty small and I figured avoiding the dependencies might make things easier. It also looked like your API wasn't geared around messages that cause long-running processes that might need multiple responses or cancellation.

If it turns out to be too much effort (it hasn't been so far; it was fairly simple getting a simple web socket open to send/receive), I might come back and give Margiebot another look, but so far things are going well!
